### PR TITLE
Adding options struct to credential types

### PR DESCRIPTION
**Pull request by catalinaperalta**
This PR adds options structs to ClientCertificateCredential.

### CODE REVIEW

**Comment by catalinaperalta** [comment]
I'm torn between adding an options bag for all credentials even the ones that dont need it at the moment or if that adds an additional (annoying?) layer when creating credentials. Thoughts?

**Comment by jhendrixMSFT** [comment]
I feel your pain, it bothers me too.  However, if we ever want to add any options later on it means we either introduce a breaking change (unlikely), or a new constructor e.g. `NewFooThingWithNewOption()` which isn't so great either (although the Go community accepts this pattern, given the limitations of the language).

Given that our design guidelines state that we use this options bag, even if it's empty, I think adding it now gives us future-proofing (we are doing this in generated code).

**Comment by jhendrixMSFT** [comment]
Personally, I would have preferred us to use variadic configuration for hand-written code as it composes well and lets us add new options.  But that ship has sailed.

**Comment by catalinaperalta** [comment]
> Given that our design guidelines state that we use this options bag, even if it's empty, I think adding it now gives us future-proofing (we are doing this in generated code).

You're right, ok I'll add the other missing options structs for the other credentials. Agreed the variadic configurations would also have been very flexible for cases like these.

File: sdk/azidentity/client_certificate_credential.go [hunk 1]
```diff
@@ -27,13 +27,28 @@ type ClientCertificateCredential struct {
...
+// password: The password required to decrypt the private key.  Pass nil if there is no password.
+// options: Manage the configuration of requests sent to Azure Active Directory.
+type ClientCertificateCredentialOptions struct {
+	// The password required to decrypt the private key.  Pass nil if there is no password.
+	Password *string
+	// Manage the configuration of requests sent to Azure Active Directory.
+	Options *TokenCredentialOptions
+}
+
+// DefaultClientCertificateCredentialOptions returns an instance of ClientCertificateCredentialOptions with the default values set.
```
Comment (jhendrixMSFT) [inline sdk/azidentity/client_certificate_credential.go:40]: "... returns an instance of ClientCertificateCredentialOptions initialized with default values."

This is the doc string I've been using across all of these `DefaultFooOptions()` funcs (comment applies to other funcs).

**Review by jhendrixMSFT — APPROVED** [review]
Few tweaks to doc strings.

File: sdk/azidentity/client_certificate_credential.go (hunk 1 above)
Comment (catalinaperalta) [reply sdk/azidentity/client_certificate_credential.go]: Done

**Pull request merged**